sinks/signalfx/signalfx.go

```diff
@@ -211,69 +213,41 @@ func (sfx *SignalFxSink) Flush(ctx context.Context, interMetrics []samplers.Inte
 	return err
 }
 
-// FlushEventsChecks sends events to SignalFx. It does not support checks.
+var successSpanTags = map[string]string{"sink": "signalfx", "results": "success"}
+var failureSpanTags = map[string]string{"sink": "signalfx", "results": "failure"}
+
+// FlushOtherSamples sends events to SignalFx. It does not support checks.
 func (sfx *SignalFxSink) FlushOtherSamples(ctx context.Context, samples []ssf.SSFSample) {
 	span, _ := trace.StartSpanFromContext(ctx, "")
 	defer span.ClientFinish(sfx.traceClient)
-
+	var countFailed = 0
+	var countSuccess = 0
 	for _, sample := range samples {
-
-		if _, ok := sample.Tags[dogstatsd.EventIdentifierKey]; !ok {
-			// This isn't an event, just continue
-			continue
-		}
-
-		dims := map[string]string{}
-
-		// Copy common dimensions in
-		for k, v := range sfx.commonDimensions {
-			dims[k] = v
-		}
-		// And hostname
-		dims[sfx.hostnameTag] = sfx.hostname
-
-		for k, v := range sample.Tags {
-			if k == dogstatsd.EventIdentifierKey {
-				// Don't copy this tag
-				continue
+		switch ddOtherSampleKind(sample.Tags) {
+		case ddSampleEvent:
+			sfx.reportEvent(ctx, &sample)
+		case ddSampleServiceCheck:
+			err := sfx.reportServiceCheck(&sample)
+			if err != nil {
+				countFailed++
+			} else {
+				countSuccess++
 			}
-			dims[k] = v
-		}
-
-		for k := range sfx.excludedTags {
-			delete(dims, k)
-		}
-		name := sample.Name
-		if len(name) > EventNameMaxLength {
-			name = name[0:EventNameMaxLength]
-		}
-		message := sample.Message
-		if len(message) > EventDescriptionMaxLength {
-			message = message[0:EventDescriptionMaxLength]
-		}
-		// Datadog requires some weird chars to do markdown… SignalFx does not so
-		// we'll chop those out
-		// https://docs.datadoghq.com/graphing/event_stream/#markdown-events
-		message = strings.Replace(message, "%%% \n", "", 1)
-		message = strings.Replace(message, "\n %%%", "", 1)
-		// Sometimes there are leading and trailing spaces
-		message = strings.TrimSpace(message)
-
-		ev := event.Event{
-			EventType:  name,
-			Category:   event.USERDEFINED,
-			Dimensions: dims,
-			Timestamp:  time.Unix(sample.Timestamp, 0),
-			Properties: map[string]interface{}{
-				"description": message,
-			},
+			// report svchk
+		default:
+			// currently only supports a service check and an event; other types are discarded
+			continue
 		}
-		// TODO: Split events out the same way as points
-		sfx.defaultClient.AddEvents(ctx, []*event.Event{&ev})
+	}
+	if countSuccess > 0 {
+		span.Add(ssf.Count(sinks.ServiceCheckConversionCount, float32(countSuccess), successSpanTags))
+	}
+	if countFailed > 0 {
+		span.Add(ssf.Count(sinks.ServiceCheckConversionCount, float32(countFailed), failureSpanTags))
```

Comment:
The coverprofile agrees - it's not executed at all by any of our tests. 
<img width="534" alt="screen shot 2018-04-11 at 2 27 59 pm" src="https://user-images.githubusercontent.com/376414/38635825-9316d9a2-3d94-11e8-93e6-4062e8ed8594.png">

Reply:
Unless my understanding of the `++` operator is wrong it'll be the number of samples for which `sfx.derivedMetrics.SendSample` failed (which may logically be 0). I think it's still a safe bet to log error counts since an error is exposed in the interface.

Reply:
Right - I was asking why we define that as part of the interface we're adding. Though it looks like we're matching the interface to a pre-existing method definition in a different package, so that's why. In that case, disregard!

Reply:
Okay. Thanks for the feedback tho. Are you asking for test coverage?

Reply:
Nope, it's fine that we're not testing our handling of an error that's being generated by a separate (pre-existing) method in a different package.

(If you want to, go ahead! But I'm fine as-is. :) )